lib/src/kvs/ds.rs

```diff
@@ -97,74 +97,104 @@ impl Datastore {
 	/// ```
 	pub async fn new(path: &str) -> Result<Datastore, Error> {
 		match path {
-			#[cfg(feature = "kv-mem")]
 			"memory" => {
-				info!(target: LOG, "Starting kvs store in {}", path);
-				let v = super::mem::Datastore::new().await.map(|v| Datastore {
-					inner: Inner::Mem(v),
-				});
-				info!(target: LOG, "Started kvs store in {}", path);
-				v
+				#[cfg(feature = "kv-mem")]
+				{
+					info!(target: LOG, "Starting kvs store in {}", path);
+					let v = super::mem::Datastore::new().await.map(|v| Datastore {
+						inner: Inner::Mem(v),
+					});
+					info!(target: LOG, "Started kvs store in {}", path);
+					v
+				}
+
+				#[cfg(not(feature = "kv-mem"))]
+				return Err(Error::Ds("Required feature `kv-mem` not enabled".to_owned()));
 			}
 			// Parse and initiate an File database
-			#[cfg(feature = "kv-rocksdb")]
 			s if s.starts_with("file:") => {
-				info!(target: LOG, "Starting kvs store at {}", path);
-				let s = s.trim_start_matches("file://");
-				let s = s.trim_start_matches("file:");
-				let v = super::rocksdb::Datastore::new(s).await.map(|v| Datastore {
-					inner: Inner::RocksDB(v),
-				});
-				info!(target: LOG, "Started kvs store at {}", path);
-				v
+				#[cfg(feature = "kv-rocksdb")]
+				{
+					info!(target: LOG, "Starting kvs store at {}", path);
+					let s = s.trim_start_matches("file://");
+					let s = s.trim_start_matches("file:");
+					let v = super::rocksdb::Datastore::new(s).await.map(|v| Datastore {
+						inner: Inner::RocksDB(v),
+					});
+					info!(target: LOG, "Started kvs store at {}", path);
+					v
+				}
+
+				#[cfg(not(feature = "kv-rocksdb"))]
+				return Err(Error::Ds("Required feature `kv-rocksdb` not enabled".to_owned()));
 			}
 			// Parse and initiate an RocksDB database
-			#[cfg(feature = "kv-rocksdb")]
 			s if s.starts_with("rocksdb:") => {
-				info!(target: LOG, "Starting kvs store at {}", path);
-				let s = s.trim_start_matches("rocksdb://");
-				let s = s.trim_start_matches("rocksdb:");
-				let v = super::rocksdb::Datastore::new(s).await.map(|v| Datastore {
-					inner: Inner::RocksDB(v),
-				});
-				info!(target: LOG, "Started kvs store at {}", path);
-				v
+				#[cfg(feature = "kv-rocksdb")]
+				{
+					info!(target: LOG, "Starting kvs store at {}", path);
+					let s = s.trim_start_matches("rocksdb://");
+					let s = s.trim_start_matches("rocksdb:");
+					let v = super::rocksdb::Datastore::new(s).await.map(|v| Datastore {
+						inner: Inner::RocksDB(v),
+					});
+					info!(target: LOG, "Started kvs store at {}", path);
+					v
+				}
+
+				#[cfg(not(feature = "kv-rocksdb"))]
+				return Err(Error::Ds("Required feature `kv-rocksdb` not enabled".to_owned()));
```

Comment:
@tobiemh Good point. How best can we make it clear that the reason it can't connect is because it's not enabled? Will this do?

'Can not connect to the `rocksdb` storage engine as it is not enabled in this build of SurrealDB.'

Reply:
Yeah that could work @rushmorem !